File: typescript/test/Wall/SpyWallDAO.ts
import {WallDAOInterface} from '../../src/Wall/WallDAOInterface';
import {User} from '../../src/User/User';
import {Brick} from '../../src/Wall/Brick';

export class SpyWallDAO implements WallDAOInterface {
    private _userParameter?: User;
    private _brickParameter?: Brick;

    get brickParameter(): Brick | undefined {
        return this._brickParameter;
    }

    get userParameter(): User | undefined {
        return this._userParameter;
    }

    getBricks(user: User): ReadonlyArray<Brick> {
        return [new Brick("", new Date())];
    }

    addBrick(user: User, brick: Brick): void {
        this._userParameter = user;
        this._brickParameter = brick;
    }
}

File: dotNet/test/Wall/SpyWallDAO.cs
using System.Collections.Immutable;
using src.Wall;

namespace test.Wall;

public class SpyWallDAO : IWallDAO
{
    public src.User.User? UserParameter { get; private set; }
    public Brick? BrickParameter { get; private set; }

    public ImmutableArray<Brick> GetBricks(src.User.User user)
    {
        return ImmutableArray.Create(new Brick("", new DateTime()));
    }

    public void AddBrick(src.User.User user, Brick brick)
    {
        UserParameter = user;
        BrickParameter = brick;
    }
}
